src/main/java/com/dreammaster/gthandler/recipes/CuttingMachineRecipes.java

```diff
@@ -23,288 +24,204 @@ public class CuttingMachineRecipes implements Runnable {
-                GT_Values.NI,
-                900,
-                1024,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_ULPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_ULPIC.get(6L),
-                GT_Values.NI,
-                900,
-                120,
-                false);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_LPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_LPIC.get(4L),
-                GT_Values.NI,
-                900,
-                480,
-                false);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_PIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_PIC.get(4L),
-                GT_Values.NI,
-                900,
-                1920,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_HPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_HPIC.get(2L),
-                GT_Values.NI,
-                900,
-                7860,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_UHPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_UHPIC.get(2L),
-                GT_Values.NI,
-                900,
-                30720,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_NPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_NPIC.get(2L),
-                GT_Values.NI,
-                900,
-                122880,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_PPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_PPIC.get(2L),
-                GT_Values.NI,
-                900,
-                500000,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_QPIC.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_QPIC.get(2L),
-                GT_Values.NI,
-                900,
-                2000000,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_NanoCPU.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_NanoCPU.get(8L),
-                GT_Values.NI,
-                900,
-                480,
-                true);
-        GT_Values.RA.addCutterRecipe(
-                ItemList.Circuit_Wafer_QuantumCPU.get(1L),
-                GT_Values.NI,
-                ItemList.Circuit_Chip_QuantumCPU.get(4L),
-                GT_Values.NI,
-                900,
-                1920,
-                true);
+
+        GT_Values.RA.stdBuilder().itemInputs(ItemList.Circuit_Wafer_ILC.get(1L))
```

Comment:
this is indeed all wrong, but the problem is different: GT_Values.RA.addCutterRecipe actually adds the 3 fluid variants. There is no recipe without fluids.

Reply:
> this is indeed all wrong, but the problem is different: GT_Values.RA.addCutterRecipe actually adds the 3 fluid variants. There is no recipe without fluids.

`sCutterRecipes` adds all of the fluid variants when there is no fluid specified aswell, but it only checks for null rather than empty inputs. It can be easily fixed in GT5 without editing anything in here, look here:

https://github.com/GTNewHorizons/GT5-Unofficial/blob/f25b1eb0f268c875c3cc2d14fe8f873f14231f46/src/main/java/gregtech/api/util/GT_Recipe.java#L2602-L2620

Reply:
Fixed in https://github.com/GTNewHorizons/GT5-Unofficial/pull/2271

![image](https://github.com/GTNewHorizons/NewHorizonsCoreMod/assets/53441451/c2f1d718-361d-4f07-bb6b-911c250d3a08)